docs/DesignMDIngestion.md

```diff
@@ -0,0 +1,61 @@
+# Generic Metadata Ingestion to MongoDB
+
+## Description
+
+This is the design document discussing the ingestion process of existing metadata
+into Zenko with MongoDB.
+
+The primary use case is to ingest existing metadata from the RING and S3 Connector;
+future use cases will include ingesting metadata from existing storage solutions
+on AWS or Azure.
+
+This specific development will allow Zenko instances to copy and ingest existing
+metadata,
+copying the information from the raft logs completely and allowing MongoDB to be
+a parallel metadata database.
+
+## Requirements
+
+* Copy all existing metadata from pre-existing storage solutions to MongoDB in Zenko.
+
+* Allow MongoDB to be used in parallel with the existing metadata servers with S3
+  Connector.
+
+## Design
+
+The proposed design will be as follows:
+
+* Connect Zenko to the S3 Connector, and Backbeat will connect to each raft session
+  and store a `log offset` (leading to total of 8 log offsets from raft log sessions
+  0 through 7).
+* Backbeat temporarily pauses real-time ingestion from the logs, and calls a listing
+  on existing items in each bucket.
+* Each object is then listed to get the metadata, which will be properly formatted
+  and sent to the queue populator in backbeat.
+* The queue populator adds the log to MongoDB.
```

Comment:
Does it mean that sending to the queue populator and adding the log are done synchronously?